Add open document operator

MohamedAbdallah17 opened the pull request:
Resolves #6.

MohamedAbdallah17 committed:
Update main.py

Implemented task : Add an open document operator 
Mohamed Abdallah

OmarEmaraDev reviewed (changes requested):
This doesn't really follow the instructions outlines in #1. It should be clear that we are reading a file, then why do you think writing a file make sense here?

MohamedAbdallah17 committed:
Update main.py

MohamedAbdallah17 committed:
Update main.py

MohamedAbdallah17 committed:
Update main.py

OmarEmaraDev commented on main.py:
```diff
@@ -107,7 +114,25 @@ def setFont(self, widget, textBuffer):
+        path = os.path.join(openDialog.get_current_folder(), openDialog.get_filename())
+        bounds = textBuffer.get_bounds()
+        deserializationFormat = textBuffer.register_deserialize_tagset()
+        deserial = textBuffer.deserialize(textBuffer, deserializationFormat, textiter, data)
```
You are currently using `data` in this invocation, but this is not defined anywhere. Data should be the bytes you read from the file. So what you should do is put this line inside the `with open(...` block, read the file data into a data variable `data = inputFile.read()`, then execute the deserialization method to get the text. Pass that text into the `set_text(text)` method of the text buffe, and you should be done.

MohamedAbdallah17 committed:
Update main.py

MohamedAbdallah17 committed:
Update main.py

here the last position i'm in .. there is an error in the deserialization function said 
" File "texteditor.py", line 132
    textBuffer.deserialize(textBuffer, deserializationFormat, startiter=textBuffer.get_start_iter(), text)
"

OmarEmaraDev reviewed (approved):
I will refactor this myself in a separate commit as needed. Thanks!
